src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -427,7 +427,7 @@ public function mirror($originDir, $targetDir, \Traversable $iterator = null, $o
     public function isAbsolutePath($file)
     {
         if (strspn($file, '/\\', 0, 1)
-            || (strlen($file) > 3 && ctype_alpha($file[0])
+            || (isset($file[3]) && ctype_alpha($file[0])
                 && substr($file, 1, 1) === ':'
```

Comment:
You could also directly access $file here instead of using substr.

(That said I have no idea when and how often those functions are called here and if those changed make a lot of difference – in our case the functions were called hundred-thousands of times and more ;))